src/types.ts

```diff
@@ -15,3 +15,12 @@ export type ButtonPanelMethods = {
   mount: (e: Element) => void
   clickHandler: (e: MouseEvent) => void
 }
+
+export type TileLoadingMonitorOptions = {
+  showLoadingThreshold: number
+  slowThreshold: number
+  onShowLoading: () => void
+  onSlowAlert: () => void
+  onLoad: () => void
+  onDestroy?: () => void
+ }
```

Comment:
Do you think put the dedicate type for some object into the object's file is better? In my view, it is more concentrated.